client/ayon_core/pipeline/load/plugins.py

```diff
@@ -433,6 +440,187 @@ def wrapped_method(self, *args, **kwargs):
+        for key in (
+            self.settings_category,
+            "load",
+            settings_name,
+        ):
+            settings = settings.get(key)
+            if settings is None:
+                self.log.debug(f"No settings found for {cls_name}")
+                return
+
+        for key, value in settings.items():
+            # Log out attributes that are not defined on plugin object
+            # - those may be potential dangerous typos in settings
+            if not hasattr(self, key):
+                self.log.debug(
+                    "Applying settings to unknown attribute"
+                    f" '{key}' on '{cls_name}'."
+                )
+            setattr(self, key, value)
+
+    @abstractmethod
+    def collect_containers(self) -> None:
+        """Collect containers from the current workfile.
+
+        This method is called by LoadContext on initialization and on reset.
+        """
+        pass
+
+    @abstractmethod
+    def is_representation_compatible(
+        self,
+        representation_context: RepresentationContext,
+    ) -> bool:
+        """Check if representation is compatible with the plugin.
+
+        Args:
+            representation_context (RepresentationContext): Representation
+                context.
+
+        Returns:
+            bool: True if compatible, False otherwise.
+
+        """
+        pass
+
+    @abstractmethod
+    def load_representations(
+        self,
+        representation_contexts: list[RepresentationContext],
+    ) -> list[ContainerItem]:
+        """Load representations.
+
+        Method still has to call 'add_containers' method on 'LoadContext'.
+
+        Args:
+            representation_contexts (list[RepresentationContext]): List of
+                representation contexts.
+
+        Returns:
+            list[ContainerItem]: List of loaded containers.
+
+        """
+        pass
+
+    @abstractmethod
+    def change_representations(
+        self,
+        items: list[tuple[ContainerItem, RepresentationContext]],
+    ) -> None:
+        """Change representations of loaded containers.
+
+        Can be used to switch between versions or different representations.
+
+        Args:
+            items (list[tuple[ContainerItem, RepresentationContext]]): List of
+                containers and their new representation contexts.
+
+        """
+        pass
+
+    @abstractmethod
+    def remove_containers(
+        self,
+        containers: list[ContainerItem]
+    ) -> None:
+        """Remove containers from the workfile.
+
+        Args:
+            containers (list[ContainerItem]): Containers to remove.
+
+        """
+
+    def can_switch_container(self, container: ContainerItem) -> bool:
+        """Check if container can be switched.
+
+        Args:
+            container (ContainerItem): Container to check.
+
+        Returns:
+            bool: True if container can be switched, False otherwise.
+
+        """
+        return False
```

Comment:
Should this include checking whether it can switch to a specific representation? Or is that supposed to be checked prior to this call?